server/channels/api4/file.go

```diff
@@ -460,7 +460,14 @@ func getFile(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 	audit.AddEventParameterAuditable(auditRec, "file", info)
 
-	if info.CreatorId != c.AppContext.Session().UserId && !c.App.SessionHasPermissionToChannelByPost(*c.AppContext.Session(), info.PostId, model.PermissionReadChannelContent) {
+	perm := false
+	if info.ChannelId != "" {
```

Comment:
Wondering if this could be simplified further. Are there are instances in which the channel ID would be empty at this point? I suppose only for non-attached files but then it wouldn't make sense to check the post either.